chain-impl-mockchain/src/fee.rs

```diff
@@ -0,0 +1,68 @@
+use crate::transaction as tx;
+use crate::value::Value;
+use chain_addr::Address;
+use std::ops::{Add, Mul};
+
+/// Linear fee using the basic affine formula `A * bytes(tx) + CONSTANT`.
+#[derive(PartialEq, Eq, PartialOrd, Debug, Clone, Copy)]
+pub struct LinearFee {
+    pub constant: Milli,
+    pub coefficient: Milli,
+}
+
+impl LinearFee {
+    pub fn new(constant: Milli, coefficient: Milli) -> Self {
+        LinearFee {
+            constant,
+            coefficient,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Debug, Clone, Copy)]
+pub struct Milli(u64);
+impl Milli {
+    pub fn new(i: u64, f: u64) -> Self {
+        Milli(i * 1000 + f % 1000)
```

Comment:
Do we need to silently overflow the fraction part with `% 1000`? Maybe we could `i * 1000 + f`?

Reply:
Not sure I just took millis from the `cardano` crate.

Reply:
the approach is kind of tied to the fact that cardano's coin are limited at 45B .. here we don't have this luxury, so simple solution is just throwing away the fractional part.

Reply:
@vincenthz I don't quite understand your comment. Why having no cap on Value makes throwing away fractional part a good idea?